server/sonar-web/src/main/js/app/components/nav/global/GlobalNavUserContainer.tsx

```diff
@@ -17,19 +17,26 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-// @flow
-import { withRouter } from 'react-router';
 import { connect } from 'react-redux';
 import GlobalNavUser from './GlobalNavUser';
 import { fetchMyOrganizations } from '../../../../apps/account/organizations/actions';
 import { getMyOrganizations } from '../../../../store/rootReducer';
 
-const mapStateToProps = state => ({
+interface StateProps {
+  organizations: Array<{ key: string; name: string }>;
+}
+
+const mapStateToProps = (state: any): StateProps => ({
   organizations: getMyOrganizations(state)
 });
 
-const mapDispatchToProps = {
-  fetchMyOrganizations
-};
+interface DispatchProps {
+  fetchMyOrganizations: () => Promise<void>;
+}
+
+// `as any`, because we don't have proper typings for redux-thunk async actions
+const mapDispatchToProps = { fetchMyOrganizations } as any;
 
-export default connect(mapStateToProps, mapDispatchToProps)(withRouter(GlobalNavUser));
+export default connect<StateProps, DispatchProps, {}>(mapStateToProps, mapDispatchToProps)(
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This is the default value for this type parameter, so it can be omitted. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS4157)